### DISCUSSION TITLE
How to verify signature with Reach wallet connector?

### DISCUSSION THREAD

**Discussion by alcyoneus-g** [reach-sh/reach-lang#1081] [Q&A]
It seems verify signature for arc14 sign-in auth is impossible with Reach. https://gist.github.com/urtho/610fb4e891f665eed2e7955e0fb3e74e

I need to produce signature and this is how it works with Randlab's myalgo connector.

```
export const signAuthTxn = async (encodedTxn): Promise<string> => {
  const txn = algosdk.decodeUnsignedTransaction(Buffer.from(encodedTxn, 'base64'));

  const myAlgoWallet = await getMyAlgo();

  const signedTxnRaw = await myAlgoWallet.signTransaction(txn.toByte());

  const encodeSTxn = window.btoa(String.fromCharCode.apply(null, signedTxnRaw?.blob as any));

  return encodeSTxn;
};
```

However, I use Reach for contract and platform will require Randlab's myalgo connector to verify signature then again Reach wallet connector to interact with contracts.

I want to avoid two different wallet connects in platform and how to make sign-in auth with Reach wallet connector?

**Comment by hmzdot** [comment]
ARC-0014 isn't merged yet, so I'd say it will be included in stdlib once it is the official way of signing messages. Currently, Reach doesn't expose a way to do that.

**Reply by alcyoneus-g** [reply]
@zetsuboii , when do you estimate to merge ARC-0014?

**Reply by hmzdot** [reply]
I am not in charge of development of Reach compiler, core team would have a more direct answer

**Reply by jeapostrophe** [reply]
ARC-0014 is something that the Algorand Foundation will review, not us --- https://github.com/algorandfoundation/ARCs/pulls --- there are already two competing versions of it.

Reach does not require that you do all things through Reach.
